### PR TITLE
ENH - refactoring network core

### PR DESCRIPTION
**Pull request by rksh**
Refactoring to remove dups and move common network static methods to ApiRequest core (out of upload)

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/ObservationUploader.java [hunk 1]
```diff
@@ -231,7 +187,7 @@ private Status doUpload( final Bundle bundle )
                         Base64.NO_WRAP) : null;
 
             // don't upload empty files
-            if ( countStats.lineCount == 0 && ! "bobzilla".equals(userName) ) {
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/ObservationUploader.java]: allow us both!

**Comment by bobzilladev** [comment]
other than our username hard coding, lgtm

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/ObservationUploader.java (hunk 1 above)
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/ObservationUploader.java]: roger! updated.

**Comment by rksh** [comment]
merging per conversation

**Pull request merged**